ssl/ssl.h

```diff
@@ -447,6 +447,15 @@ EXP_FUNC int STDCALL ssl_match_fingerprint(const SSL *ssl, const uint8_t* fp);
 EXP_FUNC int STDCALL ssl_match_spki_sha256(const SSL *ssl, const uint8_t* hash);
 
 /**
+ * @brief Check if SHA1 hash of CA public key matches the one given.
+ *
+ * @param ssl [in] An SSL object reference.
+ * @param fp [in] SHA1 hash to match against
+ * @return SSL_OK if the certificate is verified.
+ */
+EXP_FUNC int STDCALL ssl_match_auth_key_sha1(const SSL *ssl, const uint8_t* hash);
```

Comment:
I don't fully understand the usage of this function. Certificate received from the website contains an authorityKeyIdentifier, which you extract in `asn1_auth_key_id` function. The fact that a certificate contains valid authorityKeyIdentifier doesn't tell anything about authenticity of the certificate. So this function is not really useful for practical purposes of verifying the certificate.

The logic of certificate verification via authorityKeyIdentifier looks like this:
1. get authorityKeyIdentifier from the certificate
2. look up CAs RSA public key based on authorityKeyIdentifier (some map/list/etc of CA keyIdentifiers and public keys would be used in this step)
3. if authorityKeyIdentifier was not found in the map, bail out
4. if the authorityKeyIdentifier was found, use the associated public key to verify the certificate

Note that steps 1-4 happen on axTLS side; application has to provide axTLS with the map of {keyIdentifier => RSA public key} pairs. So i don't see much point in making this a public API.